cppcache/test/LoggingTest.cpp

```diff
@@ -0,0 +1,759 @@
+    auto rolledFile = fopen(rolledPath.string().c_str(), "w");
+    fwrite("Test", 1, 4, rolledFile);
+    fclose(rolledFile);
+  }
+
+  static int numOfLinesInFile(const char* fname) {
+    char line[2048];
+    char* read;
+    int ln_cnt = 0;
+    FILE* fp = fopen(fname, "r");
+    if (fp == nullptr) {
+      return 0;
+    }
+    while (!!(read = fgets(line, sizeof line, fp))) {
+      ++ln_cnt;
+    }
+
+    if (!feof(fp)) {
+      fclose(fp);
+      return -2;
+    }
+    fclose(fp);
+    return ln_cnt;
+  }
+
+  static int expected(LogLevel level) {
+    int expected = static_cast<int>(level);
+    if (level >= LogLevel::Default) {
+      expected--;
+    }
+    return expected;
+  }
+
+  static int expectedWithBanner(LogLevel level) {
+    int expected = LoggingTest::expected(level);
+    if (level != LogLevel::None) {
+      expected += LENGTH_OF_BANNER;
+    }
+    return expected;
+  }
+
+  static void verifyLineCountAtLevel(LogLevel level) {
+    apache::geode::client::Log::init(level, testLogFileName);
+
+    apache::geode::client::Log::error("Error Message");
+    apache::geode::client::Log::warning("Warning Message");
+    apache::geode::client::Log::info("Info Message");
+    apache::geode::client::Log::config("Config Message");
+    apache::geode::client::Log::fine("Fine Message");
+    apache::geode::client::Log::finer("Finer Message");
+    apache::geode::client::Log::finest("Finest Message");
+    apache::geode::client::Log::debug("Debug Message");
+
+    int lines = LoggingTest::numOfLinesInFile(testLogFileName.c_str());
+
+    ASSERT_TRUE(lines == LoggingTest::expectedWithBanner(level));
+
+    apache::geode::client::Log::close();
+    boost::filesystem::remove(testLogFileName.c_str());
+  }
+
+  static void findRolledFiles(
+      const std::string& logFilePath,
+      std::map<int32_t, boost::filesystem::path>& rolledFiles) {
+    const auto basePath =
+        boost::filesystem::absolute(boost::filesystem::path(logFilePath)) /
+        testLogFileName;
+    const auto filterstring = basePath.stem().string() + "-(\\d+)\\.log$";
+    const boost::regex my_filter(filterstring);
+
+    rolledFiles.clear();
+
+    boost::filesystem::directory_iterator end_itr;
+    for (boost::filesystem::directory_iterator i(
+             basePath.parent_path().string());
+         i != end_itr; ++i) {
+      if (boost::filesystem::is_regular_file(i->status())) {
+        std::string filename = i->path().filename().string();
+        boost::regex testPattern(filterstring);
+        boost::match_results<std::string::const_iterator> testMatches;
+        if (boost::regex_search(std::string::const_iterator(filename.begin()),
+                                filename.cend(), testMatches, testPattern)) {
+          auto index = std::atoi(
+              std::string(testMatches[1].first, testMatches[1].second).c_str());
+          rolledFiles[index] = i->path();
+        }
+      }
+    }
+  }
+
+  void testLogFnError() {
+#ifdef WIN32
```

Comment:
Is this because VC++ uses a different format for `__FUNCTION__`? It might be more appropriate to check for MSVC not WIN32 because if someone used Clang on Windows this might not hold true?

What does it look like if we switched to C++11 [`__func__`](https://en.cppreference.com/w/c/language/function_definition)?

Reply:
One of the few times I actually prefer what Microsoft has done.  __FUNCTION__ in VS is "ClassName::FunctionName", other platforms it just gives you FunctionName, which is often insufficient.  Will look into `__func__` and switch the ifdef to MSVC.

Reply:
If the née C++ 11 symbol is consistent between platforms you won't need the ifdef.

Reply:
On closer inspection, I don't think this LogFn stuff is even used. It looks like the only use is in the tests. I say delete it.

Reply:
Removed it.